spec/controllers/api/v4/questionnaires_controller_spec.rb

```diff
@@ -135,7 +135,9 @@ def create_record_list(n, options = {})
     end
 
     it "returns questionnaires with the same major dsl_version uptil given minor version" do
-      questionnaire_1 = create(:questionnaire, :active, questionnaire_type: @questionnaire_types.first, dsl_version: "1.0")
+      allow_any_instance_of(Questionnaire).to receive(:validate_layout)
```

Comment:
What is this line testing?

Reply:
Looks like this line is present in most of the tests in the PR changes. Are we trying to stub `validate_layout`? Curious why is that necessary.

Reply:
`validate_layout` only allows DSL versions `1` and `1.1`. We're mocking it's implementation whenever we need to add more DSL versions in DB for testing purpose.

`validate_layout` is mocked twice in controller and once in model test.